appinventor/buildserver/src/com/google/appinventor/buildserver/Compiler.java

```diff
@@ -504,6 +540,18 @@ private boolean writeAndroidManifest(File manifestFile, Set<String> broadcastRec
         out.write("    </activity>\n");
       }
 
+
+      // Add custom Activities
+      // make custom activities unique by putting them in one set
+      Set<String> customActivities = Sets.newHashSet();
```

Comment:
this dedup should probably happen before getting to this point; maybe in the method that creates _customActivitiesNeeded_